src/backend/gporca/libgpos/include/gpos/memory/CAutoMemoryPool.h

```diff
@@ -63,7 +63,7 @@ class CAutoMemoryPool : public CStackObject
 	CAutoMemoryPool(ELeakCheck leak_check_type = ElcExc);
 
 	// dtor
-	~CAutoMemoryPool();
+	~CAutoMemoryPool() noexcept(false);
```

Comment:
Why can't we use here `GPOS_NOEXCEPT`?

Reply:
I thought about it. The problem is that the destructor contains the method CMemoryPoolManager::GetMemoryPoolMgr()->Destroy(m_mp);. This leads to a chain of calls: -> GPOS_DELETE(mp); -> Delete(object) -> DeleteImpl(object, CMemoryPool::EatSingleton); -> DeleteImpl(ptr, eat); -> [DeleteImpl](https://github.com/arenadata/gpdb/blob/c6b7f3129ca0424f3464c5a79813f39533720c02/src/backend/gporca/libgpos/src/memory/CMemoryPoolTracker.cpp#L116).

Here, exceptions may be thrown. I am not sure if these exceptions are only used in debug mode, so I prefer to preserve the original behavior and simply mark it as noexcept(false) to be compatible with both C++98 and C++11.